step2-05/README.md

```diff
@@ -12,78 +12,80 @@ f(data) => view
 
-Facebook invented a the Flux pattern to solve this shared state issue. Redux is an implementation of the Flux architectural pattern. Redux expects the data to be a singleton state tree that listens for messages to manipulate the state as appropriate:
+Facebook invented the [Flux](https://facebook.github.io/flux/) architectural pattern to solve this shared state issue. [Redux](https://redux.js.org/) is an implementation of Flux.
+
+Redux is used inside many large, complex applications because of its clarity and predictability. It is really easy to debug and is easily extensible via its middleware architecture. In this lesson, we'll explore the heart of how Redux modifies state.
+
+Redux expects the data (store) to be a singleton state tree. It listens for messages to manipulate the state and passes updates down to views.
 
 ![Flux Diagram](../assets/flux.png)
 
-## View
+### View
 
-These are the React Components that consume the store as its data. There is a special way Redux will map its data from the state tree into the different React Components. The Components will know to re-render when these bits of state are changed.
+A view is a React components that consumes the store as its data. There is a special way Redux will map its data from the state tree into the different React components. The components will know to re-render when these bits of state are changed.
 
-## Action
+### Action
 
-Actions are messages that represent some event, such as a user's action or a network request. With the aid of _reducers_, they affect the overall state.
+Actions are messages that represent some event, such as a user's action or a network request. With the aid of *reducers*, they affect the overall state.
 
-## Store
+### Store
 
-This is a singleton state tree. The state tree is immutable and needs to be re-created at every action. This helps connected views to know when to update itself - just doing a simple reference comparison rather than a deep comparison.
+The store contains a singleton state tree. The state tree is immutable and needs to be re-created at every action. This helps connected views easily to know when to update by allowing them to do a simple reference comparison rather than a deep comparison. You can think of each state as a snapshot of the app at that point in time.
 
-## Reducers
+### Dispatcher
 
-These are simple stateless, pure functions that takes state + action message and returns a copy of state some modifications according to the action message type and payload.
+There is a single dispatcher. It simply informs the store of all the actions that need to be performed. Additional middleware (explained later) can be applied to the store, and the dispatcher's job is to dispatch the message through all the middleware layers.
 
-## Dispatcher
+### Reducers
```

Comment:
Reducers were explained multiple times, so I moved most of the content into this section